DesignSafe-CI/portal
# fix(a11y): DES-2495 new tab links in static code

## Overview: ##

Make `target="_blank"` links _either_ accessible _or_ not use `target="_blank"`.

## PR Status: ##


## Related Jira tickets: ##

* [DES-2495](https://jira.tacc.utexas.edu/browse/DES-2495)

## Summary of Changes: ##

0. Add messages with IDs for use by `{{aria-describedby}}` on `target="_blank"` links.
1. Add `{{aria-describedby}}` to `target="_blank"` links.
2. Remove `target="_blank"` from links that should not have it.*

<sup>\* I changed ones that are links within content that would be written by a CMS editor had doing so been feasible when it was created. I have not done so to links that are understandably integrated within Portal/Angular UI.</sup>

## Testing Steps: ##

1. Check that any Portal-controlled text links that open in a new window include `aria-describedby` with valid ID:

    - `msg-open-new-window`
    - `msg-open-ext-site`
    - `msg-open-ext-site-new-window`
    - `msg-open-new-window-ext-site`

2. Verify any page has hidden elements with those IDs and text representative of their meaning.

## UI Photos:

Skipped.

## Notes:

The `designsafe/templates/includes/messages.html` is assumed to load on every page, because of `{% include 'includes/messages.html' %}` on base and home templates:

<details><summary>Search Results</summary>

```
4 results - 4 files

designsafe/templates/base.j2:
  73      <div class="o-site__body">
  74:         {% include 'includes/messages.html' %}
  75          {% block breadcrumb %}{% endblock %}

designsafe/templates/cms_homepage.html:
  32  
  33:     {% include 'includes/messages.html' %}
  34  

designsafe/templates/ef_base.html:
  58      <div class="o-site__body">
  59:       {% include 'includes/messages.html' %}
  60        {% block content %}{% endblock %}

designsafe/templates/ef_cms_page.html:
  55  
  56:     {% include 'includes/messages.html' %}
  57  
```

</details>

Inspiration: https://medium.com/@svinkle/why-let-someone-know-when-a-link-opens-a-new-window-8699d20ed3b1#79b2
